src/overcooked_ai_py/mdp/overcooked_env.py

```diff
@@ -215,7 +215,10 @@ def step(self, joint_action, joint_agent_action_info=None, display_phi=False):
         """
         assert not self.is_done()
         if joint_agent_action_info is None: joint_agent_action_info = [{}, {}]
-        next_state, mdp_infos = self.mdp.get_state_transition(self.state, joint_action, display_phi, self.mp)
+        if not display_phi:
+            next_state, mdp_infos = self.mdp.get_state_transition(self.state, joint_action)
+        else:
+            next_state, mdp_infos = self.mdp.get_state_transition(self.state, joint_action, display_phi, self.mp)
```

Comment:
@nathan-miller23 let me know if this change will be OK for you. I put in this optimization so that it is not creating mp when not necessary

Reply:
Is there a reason why you wrote here `if not display_phi:` instead of `if display_phi:`? Latter seems more natural:
```
        if display_phi:
           next_state, mdp_infos = self.mdp.get_state_transition(self.state, joint_action, display_phi, self.mp)
        else:
            next_state, mdp_infos = self.mdp.get_state_transition(self.state, joint_action)
```
Anyway, it's not changing functionality and the current version is readable too so its not a big issue. I'm rather asking from curiosity.

Reply:
I see. I can update it. But I do want to get a confirmation from Nathan that this will not affect his experiments (which I believe should be the case)